tests/e2e/device/config/azure_iot_config.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZURE_IOT_CONFIG_H
+#define AZURE_IOT_CONFIG_H
+
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for AzureIoT middleware.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for AzureIoT middleware.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the AzureIoT middleware library. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "AZ IOT"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_DEBUG
+#endif
+
+/* Prototype for the function used to print to console on Windows simulator
+ * of FreeRTOS.
+ * The function prints to the console before the network is connected;
+ * then a UDP port after the network has connected. */
+extern void vLoggingPrintf( const char * pcFormatString,
+                            ... );
+
+/* Map the SdkLog macro to the logging function to enable logging
+ * on Windows simulator. */
+#ifndef SdkLog
+    #define SdkLog( message )    vLoggingPrintf message
+#endif
+
+#include "logging_stack.h"
+/************ End of logging configuration ****************/
+
+#define azureiotROOT_CA_PEM "-----BEGIN CERTIFICATE-----\r\n" \

```

Comment:
Outside scope of this PR - so feel free to resolve this comment but we should discuss further - but what's our story around helping customers get the root CA right? 

If we have some `certs.c` equivalent from C SDK, it feels the header needs a lot more comments in it (or a link to official guidance).  And if we have such an equivalent setup, ideally the e2e could reference that as well.

Reply:
Rotation of cert becomes complex, if we start distributing certs via our code. It is much better to let customer control it, we can help them to point to right cert to get via readme. Let me know what you think.

Reply:
Great point.  

Could we please put a comment in the test to this effect, like "/* This certificate is for test purposes only.  See official documentation about certificate management for your released product. */".  I don't think you actually need a link here, it's just enough to warn folks off from this one.

I sometimes go through test code to understand how stuff works and here I think we should call out this isn't the pattern - or at least there's more to understand here than copy/paste/forget.